src/charm.py

```diff
@@ -180,13 +194,19 @@ def _on_start_simulation_action(self, event: ActionEvent) -> None:
             event.fail(message=f"Failed to execute simulation: {str(e.stdout)}")
 
     def _generate_ue_config(self) -> str:
+        if self._relation_created(RFSIM_RELATION_NAME):
```

Comment:
As above, I'm afraid this can lead to an unexpected workload config if we don't create the relation on time.

Also how about breaking that `if` into `_get_sst()` and `_get_sd()`? I think it would help keeping this method clearer.

Reply:
I created the _get_ssd and get_sd methods